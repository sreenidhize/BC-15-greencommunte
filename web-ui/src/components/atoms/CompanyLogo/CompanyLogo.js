import React from "react";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  primary: {
    width: "80px",
    height: "80px",
  },
  secondary: {
    width: "50px",
    height: "50px",
  },
}));

const CompanyLogo = ({ url, alt, isPrimary }) => {
  const classes = useStyles();

  return (
    <div data-testid="img-placeholder">
      <img
        src={url}
        alt={alt}
        className={isPrimary ? classes.primary : classes.secondary}
      />
    </div>
  );
};

CompanyLogo.propTypes = {
  url: PropTypes.string,
  alt: PropTypes.string,
  isPrimary: PropTypes.bool,
};

export default CompanyLogo;
