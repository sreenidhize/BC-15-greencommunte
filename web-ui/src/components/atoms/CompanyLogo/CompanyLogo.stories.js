import React from "react";
import CompanyLogo from "./CompanyLogo";

export default {
  title: "atoms/CompanyLogo",
  component: CompanyLogo,
};

const Template = (args) => <CompanyLogo {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  url: "https://m.media-amazon.com/images/I/61b2FkbZsEL._AC_SS350_.jpg",
  alt: "BMW logo",
  isPrimary: true,
};
export const Secondary = Template.bind({});
Secondary.args = {
  url: "https://m.media-amazon.com/images/I/61b2FkbZsEL._AC_SS350_.jpg",
  alt: "BMW logo",
  isPrimary: false,
};
