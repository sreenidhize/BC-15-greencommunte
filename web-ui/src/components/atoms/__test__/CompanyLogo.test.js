import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CompanyLogo from '../CompanyLogo/CompanyLogo';

describe("CompanyLogo placeholder test", () => {
    it("Matches the snapshot", () => {
      const { container } = render(
        <CompanyLogo  />
      );
      expect(container).toMatchSnapshot();
    });
  
    test("test CompanyLogo render", () => {
      const { getByTestId } = render(<CompanyLogo  />);
      expect(getByTestId("img-placeholder")).toBeInTheDocument();
    });

  });