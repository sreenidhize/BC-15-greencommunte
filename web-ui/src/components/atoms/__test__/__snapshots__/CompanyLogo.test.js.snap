// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyLogo placeholder test Matches the snapshot 1`] = `
<div>
  <div
    data-testid="img-placeholder"
  >
    <img
      class="makeStyles-secondary-2"
    />
  </div>
</div>
`;
